DOCSP-35411 -- Add workaround for 'Mapping Potentially Null Values at Different Object Depths' to Known Issues. (#431)
mongodb/docs-bi-connector: source/reference/known-issues.txt

      GROUP BY class;
      
      Empty set (0.00 sec)

> Mapping Potentially Null Values at Different Object Depths

Reviewer: I'm thinking that this title maybe should describe the actual error (like the error above) since users are likely arriving at this page expecting to locate the issues that they're experiencing.

maybe something like "BI Connector Returns ``NULL`` Values due to Mapping Error"?

> ----------------------------------------------------------

Reviewer: Add space after header as best practice

A known mapping error can occur that causes all returned values to be ``NULL`` 

> in the result. This error occurs when you query a collection in which documents

Reviewer: A few small suggestions here
- can we use active voice here?
-  it's implied that we're referring to the documents in the collection, so we can remove that bit of extra language
- If you decide to change the title to be more focused on the error itself, we can maybe include that high-level language about object depths

Maybe something like:

> 
> A known mapping error can occur that causes |bi-short| to return only ``NULL``
> values. This error occurs when you map potentially ``NULL`` values at
> different object depths. Specifically, the error occurs when you query a 
> collection that contains.... etc

> contain an array field that can be ``NULL`` and that array contains 

Reviewer: "that" is a little confusing here as a construction. To clarify:

```suggestion
contain an array field that can be ``NULL,`` and the same array contains 
```

an additional field that can also be ``NULL``. A workaround for this error is to 

> include the ``nonnullsched`` parameter with the ``createView`` method.

Reviewer: Can we revise this to add context on how this workaround actually works?
Perhaps:

To resolve this error, create a view that contains only documents where the array field is an array
by including the ``nonnullsched`` parameter with the ``createView`` method.

.. example::

>    For a collection named ``nextDeparture`` with an array field that is 
>    sometimes ``NULL`` called ``response.sechedule``, we can create the 

Reviewer: Reframe for clarity (+ typo fix)

```suggestion
   For a collection named ``nextDeparture`` with an array field named 
   ``response.schedule`` that is sometimes ``NULL``, you can create the 
```

      GROUP BY class;
      
      Empty set (0.00 sec)

> BI Connector Returns ``NULL`` Values due to Mapping Error

Reviewer: ```suggestion
BI Connector Returns ``NULL`` Values Due to Mapping Error
```

----------------------------------------------------------

> A known mapping error can occur that causes all returned values to be ``NULL`` 
> in the result. This error occurs when you query a collection that

Reviewer: active voice
```suggestion
A known mapping error can occur that causes |bi-short| to return only ``NULL``
values. This error occurs when you query a collection that
```

contains an array field that can be ``NULL``, and the same array contains 
an additional field that can also be ``NULL``. A workaround for this error is to 

> create a view by including the ``nonnullsched`` parameter with the ``createView`` method.

Reviewer: Can you add some context on what this view returns and why this is a valid workaround? From the ticket description:

> The workaround is to create a view that contains only documents where the array field is an array:

      GROUP BY class;
      
      Empty set (0.00 sec)

BI Connector Returns ``NULL`` Values Due to Mapping Error
----------------------------------------------------------

A known mapping error can occur that causes |bi-short| to return all ``NULL`` 
values. This error occurs when you query a collection that
contains an array field that can be ``NULL``, and the same array contains 
an additional field that can also be ``NULL``. A workaround for this error is to 

> create a view by including the ``nonnullsched`` parameter with the 

Reviewer 2: nonnullsched is the name of the view in this case. The important thing is the match that filters on null :)
